fieldpath/set_test.go

```diff
@@ -755,3 +756,255 @@ func TestSetNodeMapIterate(t *testing.T) {
+						{Name: "key2", Value: value.NewValueInterface("valueY")},
+					}, "field"),
+			),
+			filter: NewIncludeMatcherFilter(MakePrefixMatcherOrDie("spec", "listMap", &value.FieldList{
+				{Name: "key1", Value: value.NewValueInterface("value1")},
+				{Name: "key2", Value: value.NewValueInterface("value2")},
+			})),
+			expect: NewSet(
+				MakePathOrDie("spec"),
+				MakePathOrDie("spec", "listMap",
+					&value.FieldList{
+						{Name: "key1", Value: value.NewValueInterface("value1")},
+						{Name: "key2", Value: value.NewValueInterface("value2")},
+					}),
+				MakePathOrDie("spec", "listMap",
+					&value.FieldList{
+						{Name: "key1", Value: value.NewValueInterface("value1")},
+						{Name: "key2", Value: value.NewValueInterface("value2")},
+					}, "field"),
+			),
+		},
+		{
+			name: "filter listMap key",
+			input: NewSet(
+				MakePathOrDie("spec"),
+				MakePathOrDie("spec", "listMap",
+					&value.FieldList{
+						{Name: "key1", Value: value.NewValueInterface("value1")},
+						{Name: "key2", Value: value.NewValueInterface("value2")},
+					}),
+				MakePathOrDie("spec", "listMap",
+					&value.FieldList{
+						{Name: "key1", Value: value.NewValueInterface("value1")},
+						{Name: "key2", Value: value.NewValueInterface("value2")},
+					}, "field"),
+				MakePathOrDie("spec", "listMap",
+					&value.FieldList{
+						{Name: "key1", Value: value.NewValueInterface("valueX")},
+						{Name: "key2", Value: value.NewValueInterface("valueY")},
+					}, "field"),
+			),
+			filter: NewIncludeMatcherFilter(MakePrefixMatcherOrDie("spec", "listMap", &value.FieldList{
+				{Name: "key1", Value: value.NewValueInterface("value1")},
+				{Name: "key2", Value: value.NewValueInterface("value2")},
+			})),
+			expect: NewSet(
+				MakePathOrDie("spec"),
+				MakePathOrDie("spec", "listMap",
+					&value.FieldList{
+						{Name: "key1", Value: value.NewValueInterface("value1")},
+						{Name: "key2", Value: value.NewValueInterface("value2")},
+					}),
+				MakePathOrDie("spec", "listMap",
+					&value.FieldList{
+						{Name: "key1", Value: value.NewValueInterface("value1")},
+						{Name: "key2", Value: value.NewValueInterface("value2")},
+					}, "field"),
+			),
+		},
+		{
+			name: "filter value",
+			input: NewSet(
+				MakePathOrDie("spec"),
+				MakePathOrDie("spec", "set", value.NewValueInterface("v1")),
+				MakePathOrDie("spec", "set", value.NewValueInterface("v2")),
+			),
+			filter: NewIncludeMatcherFilter(MakePrefixMatcherOrDie("spec", "set", value.NewValueInterface("v1"))),
+			expect: NewSet(
+				MakePathOrDie("spec"),
+				MakePathOrDie("spec", "set", value.NewValueInterface("v1")),
+			),
+		},
+		{
+			name: "filter by index",
+			input: NewSet(
+				MakePathOrDie("spec"),
+				MakePathOrDie("spec", "list", 0, "value"),
+				MakePathOrDie("spec", "list", 1, "value"),
+			),
+			filter: NewIncludeMatcherFilter(MakePrefixMatcherOrDie("spec", "list", 1, "value")),
+			expect: NewSet(
+				MakePathOrDie("spec"),
```

Comment:
Unclear to me why this has "spec" and "spec.list[1].value" but not "spec.list" and "spec.list[1]" as intermediate steps.  But "wildcard takes precedence" case below does not match "spec" ?

Reply:
I was just inconsistent about what included in the inputs.

I've done a sweep to add all parents to inputs.  Hopefully this helps clarify what exactly exact match does.